apps/web/components/eventtype/Locations.tsx

```diff
@@ -268,23 +268,6 @@ const Locations: React.FC<LocationsProps> = ({
                         });
                         showToast(t("location_already_exists"), "warning");
                       }
-                      // Whenever location changes, we need to reset the locations item in booking questions list else it overflows
-                      // previously added values resulting in wrong behaviour
-                      const existingBookingFields = getValues("bookingFields");
-                      const findLocation = existingBookingFields.findIndex(
-                        (field) => field.name === "location"
-                      );
-                      if (findLocation >= 0) {
-                        existingBookingFields[findLocation] = {
-                          ...existingBookingFields[findLocation],
-                          type: "radioInput",
-                          label: "",
-                          placeholder: "",
-                        };
-                        setValue("bookingFields", existingBookingFields, {
-                          shouldDirty: true,
-                        });
-                      }
```

Comment:
Reverting the code in the faulty PR. No new addition here.

FormBuilder has a dataStore property by which it is made aware of locations and then FormBuilder or the component using the FormBuilder component should do any transformations.